gasstation/gasstattion.go

```diff
@@ -54,11 +66,8 @@ func (gs *GasStation) SuggestGasPrice() (uint64, error) {
 
 		smallestPrice := blk.Actions[0].GasPrice()
```

Comment:
This is problematic, because blk.Actions[0] could be a system action.

So it would happen that in a block there's just one GrantReward action, and no user action. In this case, you should have nil smallestPrice.